obs-studio-server/source/nodeobs_settings.cpp

```diff
@@ -1010,6 +1010,13 @@ void OBS_settings::saveStreamSettings(std::vector<SubCategory> streamSettings)
 
 	obs_data_release(hotkeyData);
 
+	if (OBS_service::isStreamingOutputActive()) {
+		blog(
+		    LOG_WARNING,
+		    "Do not create a new service while old '%s' service is active and streaming",
+		    currentServiceName);
```

Comment:
Should we do more than just adding a log here?

Reply:
I added logs as a 'downgrade' from this PR: https://github.com/stream-labs/obs-studio-node/pull/622 ;

The problem was here is that frontend invoked `saveStreamSettings `during streaming, which is incorrect (the fix is here - https://github.com/stream-labs/streamlabs-obs/pull/2447 ). Thus, this log was added to track incorrect usage of backend when this crash will happen again.

Reply:
I see, there should be no issue if we also return in the backend in case a call is indeed made from the frontend while streaming?

Reply:
from my perspective - there are shouldn't be any issue. However, as I remember, @summeroff was against this option to silently return.

Reply:
In my opinion, we can't just return, we had to fix it in the frontend since the logic was wrong there. Now nothing prevents us to make sure our code doesn't crash if the frontend starts re doing the same thing in the future.

Reply:
I think that it is better to return error to frontend where it can be recognized and fixed, waiting for crash reports will make it longer and user experience worse. 

It can be done in separate PR as it is not a main issue of this PR.

Reply:
Agree, I would like to see the returning as part of this PR as a first step. Please make the change.

Reply:
Added return. Also, moved this guard to the top of the function to protect from torn state of stream settings:

Reply:
Perfect